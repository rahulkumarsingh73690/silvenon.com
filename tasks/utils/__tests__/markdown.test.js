const { processor } = require('../markdown')
const dedent = require('dedent')

jest.mock('../image-size', () => async () => ({
  width: 800,
  height: 600,
}))

describe('Markdown processor', () => {
  test('output', async () => {
    const result = await processor.process(dedent`
      Introduction to my post with smart -- punctuation.

      ## Section

      <img
        alt=""
        data-cloudinary-id="my-image"
        data-unsplash
        data-name="Matija MarohniÄ‡"
        data-username="silvenon">

      ~~~js{2}
      function doSomething() {
        console.log('hello')
      }
      ~~~

      [[tip | Hot tip!]]
      | Content.
      |
      | More content.
    `)
    const template = document.createElement('template')
    template.innerHTML = result.contents
    expect(template.content).toMatchSnapshot()
  })
})
