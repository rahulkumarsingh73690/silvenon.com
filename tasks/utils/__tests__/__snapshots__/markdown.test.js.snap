// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown processor output 1`] = `
<DocumentFragment>
  <p>
    Introduction to my post with smart — punctuation.
  </p>
  

  <h2>
    Section
  </h2>
  

  <p />
  <figure>
    <div
      class="ar"
      style="--w: 800; --h: 600; max-width: 400px;"
    >
      <div
        class="ar-media"
      >
        <img
          alt=""
          loading="lazy"
          sizes="(min-width: 1536px) 952.91px, (min-width: 1280px) 811.865px, (min-width: 1024px) 735.82px, (min-width: 640px) 655.078px, calc(100vw - 2rem)"
          src="https://res.cloudinary.com/silvenon/image/upload/c_limit/my-image"
          srcset="https://res.cloudinary.com/silvenon/image/upload/c_limit,w_640/my-image 640w, https://res.cloudinary.com/silvenon/image/upload/c_limit,w_768/my-image 768w"
        />
      </div>
    </div>
    <figcaption>
      <span>
        Photo by 
      </span>
      <a
        href="https://unsplash.com/@silvenon?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText"
      >
        Matija Marohnić
      </a>
      <span>
         on 
      </span>
      <a
        href="https://unsplash.com/?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText"
      >
        Unsplash
      </a>
    </figcaption>
  </figure>
  <p />
  

  <pre
    class="language-js"
  >
    <code
      class="language-js{2}"
    >
      <span
        class="token keyword"
      >
        function
      </span>
       
      <span
        class="token function"
      >
        doSomething
      </span>
      <span
        class="token punctuation"
      >
        (
      </span>
      <span
        class="token punctuation"
      >
        )
      </span>
       
      <span
        class="token punctuation"
      >
        {
      </span>
      

      <div
        class="mdx-marker"
      >
          
        <span
          class="token console class-name"
        >
          console
        </span>
        <span
          class="token punctuation"
        >
          .
        </span>
        <span
          class="token method function property-access"
        >
          log
        </span>
        <span
          class="token punctuation"
        >
          (
        </span>
        <span
          class="token string"
        >
          'hello'
        </span>
        <span
          class="token punctuation"
        >
          )
        </span>
        

      </div>
      <span
        class="token punctuation"
      >
        }
      </span>
      

    </code>
  </pre>
  

  <div
    class="custom-block tip"
  >
    <div
      class="custom-block-heading"
    >
      Hot tip!
    </div>
    <div
      class="custom-block-body"
    >
      <p>
        Content.
      </p>
      <p>
        More content.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
